base/essentials.jl

```diff
@@ -212,7 +212,9 @@ function tuple_type_tail(T::Type)
     else
         T.name === Tuple.name || throw(MethodError(tuple_type_tail, (T,)))
         if isvatuple(T) && length(T.parameters) == 1
-            return T
+            va = T.parameters[1]
```

Comment:
```suggestion
            va = unwrap_va(T.parameters[end])
```
(but don't forget you also need to rewrap it)

Reply:
I don't really want to unwrap it though. If `N` is UnionAll'ed here, I just want to return it.